sky/packages/sky/lib/widgets/drawer.dart

```diff
@@ -34,12 +39,47 @@ const double _kWidth = 304.0;
 const double _kMinFlingVelocity = 365.0;
 const double _kFlingVelocityScale = 1.0 / 300.0;
 const Duration _kBaseSettleDuration = const Duration(milliseconds: 246);
+const Duration _kThemeChangeDuration = const Duration(milliseconds: 200);
 const Point _kOpenPosition = Point.origin;
 const Point _kClosedPosition = const Point(-_kWidth, 0.0);
 
 typedef void DrawerStatusChangedCallback(AnimationStatus status);
 
-class Drawer extends AnimatedComponent {
+class ColorTransitionIntention extends AnimationIntention {
```

Comment:
Agreed - along with the SlideInIntention. I'll move them after you review so it'll be easier to see what changed.